src/olympia/blocklist/mlbf.py

```diff
@@ -136,25 +171,14 @@ def __init__(self, *args, **kwargs):
 
     @cached_property
     def _all_blocks(self):
-        return (
-            BlockVersion.objects.filter(version__file__is_signed=True)
-            .distinct()
-            .order_by('id')
-            .values_list(
-                'block__guid',
-                'version__version',
-                'version_id',
-                'block_type',
-                named=True,
-            )
-        )
+        return get_all_blocked_items()
 
     def _format_blocks(self, block_type: BlockType) -> List[str]:
         return MLBF.hash_filter_inputs(
             [
-                (version.block__guid, version.version__version)
+                (version[0], version[1])
                 for version in self._all_blocks
-                if version.block_type == block_type
+                if version[3] == block_type
```

Comment:
could you switch back to _named_ values? or what's the problem otherwise?

Reply:
I can for this patch, but in order to mock the result using named values means I need to either create millions of BlockVersion instances (expensive) or create a custom class with these properties (expensive and ugly)

Reply:
I suppose a `@dataclass` might be a decent middle ground since I'd rather keep the production code easier to read than less obvious because of the tests. Let's continue the conversation in the follow-up patch.